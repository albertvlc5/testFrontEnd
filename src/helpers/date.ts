export const getPreviousMonthNameByLocale = (
  dateStr: string,
  locale = 'pt-BR',
) => {
  const date = new Date(dateStr);
  const month = date.getUTCMonth();
  const previousMonth = month === 0 ? 11 : month - 1;

  const monthName = new Date(
    date.getUTCFullYear(),
    previousMonth,
    1,
  ).toLocaleString(locale, { month: 'long' });

  return monthName;
};
