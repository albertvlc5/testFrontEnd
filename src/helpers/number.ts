import 'intl';
import 'intl/locale-data/jsonp/pt';
export const formatNumberByLocale = (value: number, locale = 'pt-Br'): string =>
  new Intl.NumberFormat(locale, { minimumFractionDigits: 2 }).format(value);

export const formatMoney2Numeral = (money: string): number => {
  const [value, cents] = String(money).split(',');
  return Number([value.replace(/\D+/g, ''), cents].join('.'));
};

export const numberToCurrencyIntl = (
  value: number = 0,
  locale = 'pt-BR',
  currency = 'BRL',
) =>
  new Intl.NumberFormat(locale, { style: 'currency', currency }).format(value);
