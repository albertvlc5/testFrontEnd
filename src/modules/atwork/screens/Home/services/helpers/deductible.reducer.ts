import {
  IDeductible,
  IDeductibleTransformed,
} from '../../interfaces/deductible.interface';

export const initialDeductibleTransformed = {
  totalDiscounts: 0,
  dueDate: '',
};

export const deductibleReducer = (
  acumulator: IDeductibleTransformed,
  item: IDeductible,
) => ({
  totalDiscounts: acumulator.totalDiscounts + item.amountDeductibleThisMonth,
  dueDate: item.dueDate,
});
