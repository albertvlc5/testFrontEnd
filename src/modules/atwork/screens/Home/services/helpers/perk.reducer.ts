import { ProductType } from '../../interfaces/product-type.enum';
import { IPerk, IPerkTransformed } from '../../interfaces/perk.interface';

export const initialPerkTransformed = {
  totalAmount: 0,
  salaryAdvancement: {
    totalAmountAvailable: 0,
    amountDeductible: 0,
    currency: '',
  },
  payroll: {
    amountDeductible: undefined,
    totalAmountAvailable: 0,
    currency: '',
  },
};

export const perkReducer = (acumulator: IPerkTransformed, item: IPerk) => {
  const salaryAdvancement =
    item.type === ProductType.SALARY_ADVANCEMENT
      ? {
          totalAmountAvailable: item.totalAmountAvailable,
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
        }
      : acumulator.salaryAdvancement;

  const payroll =
    item.type === ProductType.PAYROLL
      ? {
          amountDeductible: item.amountDeductible,
          currency: item.symbol,
          totalAmountAvailable: item.totalAmountAvailable,
        }
      : acumulator.payroll;

  return {
    totalAmount: acumulator.totalAmount + item.totalAmountAvailable,
    salaryAdvancement: salaryAdvancement,
    payroll: payroll,
  };
};
