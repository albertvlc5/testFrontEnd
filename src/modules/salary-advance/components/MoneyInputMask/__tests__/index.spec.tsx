import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { mountWithAtworkHost } from '../../../../../__tests__/helpers';
import { MoneyInputMask } from '../index';

describe('Money Input Mask', () => {
  it('should render component', () => {
    const wrapper = mountWithAtworkHost(
      <MoneyInputMask
        currency="BRL"
        onChangeText={jest.fn()}
        value={'4.250,00'}
      />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with BRL currency format', () => {
    const onChangeMock = jest.fn();
    const { getByTestId } = mountWithAtworkHost(
      <MoneyInputMask
        currency="BRL"
        onChangeText={onChangeMock}
        value={'22'}
      />,
    );
    const comp = getByTestId('input-mask-test-id');
    fireEvent.changeText(comp, '50.00');
    expect(onChangeMock).toHaveBeenCalledWith('50,00', undefined);
  });
});
