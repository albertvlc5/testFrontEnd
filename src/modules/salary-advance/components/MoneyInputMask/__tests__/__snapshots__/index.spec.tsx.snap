// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money Input Mask should render component 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    color="neutral5"
    strikeThrough={false}
    style={
      Array [
        Object {
          "color": "#aabbbb",
          "fontFamily": "HelveticaNowDisplay-Light",
          "fontSize": 16,
          "lineHeight": 20,
          "textDecorationColor": "#aabbbb",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        },
        Object {
          "fontSize": 32,
          "marginRight": 15,
          "paddingTop": 27,
        },
      ]
    }
    underlined={false}
    variant="mockTokenHelveticaDisplayLight"
  >
    R$
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    options={
      Object {
        "maskType": "BRL",
        "unit": "",
      }
    }
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "fontFamily": "HelveticaNowDisplay-Light",
          "fontSize": 32,
          "letterSpacing": 0.6,
          "lineHeight": 40,
          "width": "100%",
        },
      ]
    }
    testID="input-mask-test-id"
    type="money"
    underlineColorAndroid="transparent"
    value="4.250,00"
  />
</View>
`;
